// @aws-sdk/client-geo-places/dist-cjs/models/models_0.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AutocompleteAddressHighlightsFilterSensitiveLog = exports.SubRegionHighlightsFilterSensitiveLog = exports.RegionHighlightsFilterSensitiveLog = exports.CountryHighlightsFilterSensitiveLog = exports.HighlightFilterSensitiveLog = exports.AutocompleteRequestFilterSensitiveLog = exports.AutocompleteFilterFilterSensitiveLog = exports.FilterCircleFilterSensitiveLog = exports.AddressComponentPhonemesFilterSensitiveLog = exports.PhonemeTranscriptionFilterSensitiveLog = exports.AddressFilterSensitiveLog = exports.SubRegionFilterSensitiveLog = exports.StreetComponentsFilterSensitiveLog = exports.SecondaryAddressComponentFilterSensitiveLog = exports.RegionFilterSensitiveLog = exports.CountryFilterSensitiveLog = exports.AccessRestrictionFilterSensitiveLog = exports.CategoryFilterSensitiveLog = exports.AccessPointFilterSensitiveLog = exports.SuggestResultItemType = exports.QueryType = exports.SuggestIntendedUse = exports.SuggestAdditionalFeature = exports.SearchTextIntendedUse = exports.SearchTextAdditionalFeature = exports.SearchNearbyIntendedUse = exports.SearchNearbyAdditionalFeature = exports.ReverseGeocodeIntendedUse = exports.ReverseGeocodeFilterPlaceType = exports.ReverseGeocodeAdditionalFeature = exports.GetPlaceIntendedUse = exports.GetPlaceAdditionalFeature = exports.RecordTypeCode = exports.ZipClassificationCode = exports.PostalCodeType = exports.PostalAuthority = exports.GeocodeIntendedUse = exports.GeocodeFilterPlaceType = exports.GeocodeAdditionalFeature = exports.ValidationException = exports.ValidationExceptionReason = exports.ThrottlingException = exports.InternalServerException = exports.PlaceType = exports.PostalCodeMode = exports.AutocompleteIntendedUse = exports.AutocompleteFilterPlaceType = exports.AutocompleteAdditionalFeature = exports.TypePlacement = exports.AccessDeniedException = void 0;
exports.SuggestResponseFilterSensitiveLog = exports.SuggestResultItemFilterSensitiveLog = exports.SuggestQueryResultFilterSensitiveLog = exports.SuggestPlaceResultFilterSensitiveLog = exports.SuggestHighlightsFilterSensitiveLog = exports.SuggestAddressHighlightsFilterSensitiveLog = exports.QueryRefinementFilterSensitiveLog = exports.SuggestRequestFilterSensitiveLog = exports.SuggestFilterFilterSensitiveLog = exports.SearchTextResponseFilterSensitiveLog = exports.SearchTextResultItemFilterSensitiveLog = exports.SearchTextRequestFilterSensitiveLog = exports.SearchTextFilterFilterSensitiveLog = exports.SearchNearbyResponseFilterSensitiveLog = exports.SearchNearbyResultItemFilterSensitiveLog = exports.SearchNearbyRequestFilterSensitiveLog = exports.SearchNearbyFilterFilterSensitiveLog = exports.ReverseGeocodeResponseFilterSensitiveLog = exports.ReverseGeocodeResultItemFilterSensitiveLog = exports.ReverseGeocodeRequestFilterSensitiveLog = exports.GetPlaceResponseFilterSensitiveLog = exports.PhonemeDetailsFilterSensitiveLog = exports.OpeningHoursFilterSensitiveLog = exports.OpeningHoursComponentsFilterSensitiveLog = exports.GetPlaceRequestFilterSensitiveLog = exports.GeocodeResponseFilterSensitiveLog = exports.GeocodeResultItemFilterSensitiveLog = exports.TimeZoneFilterSensitiveLog = exports.PostalCodeDetailsFilterSensitiveLog = exports.UspsZipPlus4FilterSensitiveLog = exports.UspsZipFilterSensitiveLog = exports.GeocodeParsedQueryFilterSensitiveLog = exports.GeocodeParsedQueryAddressComponentsFilterSensitiveLog = exports.ParsedQuerySecondaryAddressComponentFilterSensitiveLog = exports.ParsedQueryComponentFilterSensitiveLog = exports.RelatedPlaceFilterSensitiveLog = exports.IntersectionFilterSensitiveLog = exports.GeocodeRequestFilterSensitiveLog = exports.GeocodeQueryComponentsFilterSensitiveLog = exports.GeocodeFilterFilterSensitiveLog = exports.FoodTypeFilterSensitiveLog = exports.ContactsFilterSensitiveLog = exports.ContactDetailsFilterSensitiveLog = exports.BusinessChainFilterSensitiveLog = exports.AutocompleteResponseFilterSensitiveLog = exports.AutocompleteResultItemFilterSensitiveLog = exports.AutocompleteHighlightsFilterSensitiveLog = void 0;
const smithy_client_1 = require("@smithy/smithy-client");
const GeoPlacesServiceException_1 = require("./GeoPlacesServiceException");
class AccessDeniedException extends GeoPlacesServiceException_1.GeoPlacesServiceException {
    name = "AccessDeniedException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "AccessDeniedException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, AccessDeniedException.prototype);
        this.Message = opts.Message;
    }
}
exports.AccessDeniedException = AccessDeniedException;
exports.TypePlacement = {
    AFTER_BASE_NAME: "AfterBaseName",
    BEFORE_BASE_NAME: "BeforeBaseName",
};
exports.AutocompleteAdditionalFeature = {
    CORE: "Core",
};
exports.AutocompleteFilterPlaceType = {
    LOCALITY: "Locality",
    POSTAL_CODE: "PostalCode",
};
exports.AutocompleteIntendedUse = {
    SINGLE_USE: "SingleUse",
};
exports.PostalCodeMode = {
    ENUMERATE_SPANNED_LOCALITIES: "EnumerateSpannedLocalities",
    MERGE_ALL_SPANNED_LOCALITIES: "MergeAllSpannedLocalities",
};
exports.PlaceType = {
    BLOCK: "Block",
    COUNTRY: "Country",
    DISTRICT: "District",
    INTERPOLATED_ADDRESS: "InterpolatedAddress",
    INTERSECTION: "Intersection",
    LOCALITY: "Locality",
    POINT_ADDRESS: "PointAddress",
    POINT_OF_INTEREST: "PointOfInterest",
    POSTAL_CODE: "PostalCode",
    REGION: "Region",
    SECONDARY_ADDRESS: "SecondaryAddress",
    STREET: "Street",
    SUB_BLOCK: "SubBlock",
    SUB_DISTRICT: "SubDistrict",
    SUB_REGION: "SubRegion",
};
class InternalServerException extends GeoPlacesServiceException_1.GeoPlacesServiceException {
    name = "InternalServerException";
    $fault = "server";
    $retryable = {};
    Message;
    constructor(opts) {
        super({
            name: "InternalServerException",
            $fault: "server",
            ...opts,
        });
        Object.setPrototypeOf(this, InternalServerException.prototype);
        this.Message = opts.Message;
    }
}
exports.InternalServerException = InternalServerException;
class ThrottlingException extends GeoPlacesServiceException_1.GeoPlacesServiceException {
    name = "ThrottlingException";
    $fault = "client";
    $retryable = {};
    Message;
    constructor(opts) {
        super({
            name: "ThrottlingException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ThrottlingException.prototype);
        this.Message = opts.Message;
    }
}
exports.ThrottlingException = ThrottlingException;
exports.ValidationExceptionReason = {
    CANNOT_PARSE: "CannotParse",
    FIELD_VALIDATION_FAILED: "FieldValidationFailed",
    MISSING: "Missing",
    OTHER: "Other",
    UNKNOWN_FIELD: "UnknownField",
    UNKNOWN_OPERATION: "UnknownOperation",
};
class ValidationException extends GeoPlacesServiceException_1.GeoPlacesServiceException {
    name = "ValidationException";
    $fault = "client";
    Message;
    Reason;
    FieldList;
    constructor(opts) {
        super({
            name: "ValidationException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ValidationException.prototype);
        this.Message = opts.Message;
        this.Reason = opts.Reason;
        this.FieldList = opts.FieldList;
    }
}
exports.ValidationException = ValidationException;
exports.GeocodeAdditionalFeature = {
    ACCESS: "Access",
    INTERSECTIONS: "Intersections",
    SECONDARY_ADDRESSES: "SecondaryAddresses",
    TIME_ZONE: "TimeZone",
};
exports.GeocodeFilterPlaceType = {
    INTERPOLATED_ADDRESS: "InterpolatedAddress",
    INTERSECTION: "Intersection",
    LOCALITY: "Locality",
    POINT_ADDRESS: "PointAddress",
    POSTAL_CODE: "PostalCode",
    STREET: "Street",
};
exports.GeocodeIntendedUse = {
    SINGLE_USE: "SingleUse",
    STORAGE: "Storage",
};
exports.PostalAuthority = {
    USPS: "Usps",
};
exports.PostalCodeType = {
    USPS_ZIP: "UspsZip",
    USPS_ZIP_PLUS_4: "UspsZipPlus4",
};
exports.ZipClassificationCode = {
    MILITARY: "Military",
    POST_OFFICE_BOXES: "PostOfficeBoxes",
    UNIQUE: "Unique",
};
exports.RecordTypeCode = {
    FIRM: "Firm",
    GENERAL: "General",
    HIGH_RISE: "HighRise",
    POST_OFFICE_BOX: "PostOfficeBox",
    RURAL: "Rural",
    STREET: "Street",
};
exports.GetPlaceAdditionalFeature = {
    ACCESS: "Access",
    CONTACT: "Contact",
    PHONEMES: "Phonemes",
    SECONDARY_ADDRESSES: "SecondaryAddresses",
    TIME_ZONE: "TimeZone",
};
exports.GetPlaceIntendedUse = {
    SINGLE_USE: "SingleUse",
    STORAGE: "Storage",
};
exports.ReverseGeocodeAdditionalFeature = {
    ACCESS: "Access",
    INTERSECTIONS: "Intersections",
    TIME_ZONE: "TimeZone",
};
exports.ReverseGeocodeFilterPlaceType = {
    INTERPOLATED_ADDRESS: "InterpolatedAddress",
    INTERSECTION: "Intersection",
    LOCALITY: "Locality",
    POINT_ADDRESS: "PointAddress",
    STREET: "Street",
};
exports.ReverseGeocodeIntendedUse = {
    SINGLE_USE: "SingleUse",
    STORAGE: "Storage",
};
exports.SearchNearbyAdditionalFeature = {
    ACCESS: "Access",
    CONTACT: "Contact",
    PHONEMES: "Phonemes",
    TIME_ZONE: "TimeZone",
};
exports.SearchNearbyIntendedUse = {
    SINGLE_USE: "SingleUse",
    STORAGE: "Storage",
};
exports.SearchTextAdditionalFeature = {
    ACCESS: "Access",
    CONTACT: "Contact",
    PHONEMES: "Phonemes",
    TIME_ZONE: "TimeZone",
};
exports.SearchTextIntendedUse = {
    SINGLE_USE: "SingleUse",
    STORAGE: "Storage",
};
exports.SuggestAdditionalFeature = {
    ACCESS: "Access",
    CORE: "Core",
    PHONEMES: "Phonemes",
    TIME_ZONE: "TimeZone",
};
exports.SuggestIntendedUse = {
    SINGLE_USE: "SingleUse",
};
exports.QueryType = {
    BUSINESS_CHAIN: "BusinessChain",
    CATEGORY: "Category",
};
exports.SuggestResultItemType = {
    PLACE: "Place",
    QUERY: "Query",
};
const AccessPointFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Position && { Position: smithy_client_1.SENSITIVE_STRING }),
});
exports.AccessPointFilterSensitiveLog = AccessPointFilterSensitiveLog;
const CategoryFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Id && { Id: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Name && { Name: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.LocalizedName && { LocalizedName: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Primary && { Primary: smithy_client_1.SENSITIVE_STRING }),
});
exports.CategoryFilterSensitiveLog = CategoryFilterSensitiveLog;
const AccessRestrictionFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Restricted && { Restricted: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Categories && { Categories: obj.Categories.map((item) => (0, exports.CategoryFilterSensitiveLog)(item)) }),
});
exports.AccessRestrictionFilterSensitiveLog = AccessRestrictionFilterSensitiveLog;
const CountryFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Code2 && { Code2: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Code3 && { Code3: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Name && { Name: smithy_client_1.SENSITIVE_STRING }),
});
exports.CountryFilterSensitiveLog = CountryFilterSensitiveLog;
const RegionFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Code && { Code: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Name && { Name: smithy_client_1.SENSITIVE_STRING }),
});
exports.RegionFilterSensitiveLog = RegionFilterSensitiveLog;
const SecondaryAddressComponentFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Number && { Number: smithy_client_1.SENSITIVE_STRING }),
});
exports.SecondaryAddressComponentFilterSensitiveLog = SecondaryAddressComponentFilterSensitiveLog;
const StreetComponentsFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.BaseName && { BaseName: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Type && { Type: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Prefix && { Prefix: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Suffix && { Suffix: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Direction && { Direction: smithy_client_1.SENSITIVE_STRING }),
});
exports.StreetComponentsFilterSensitiveLog = StreetComponentsFilterSensitiveLog;
const SubRegionFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Code && { Code: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Name && { Name: smithy_client_1.SENSITIVE_STRING }),
});
exports.SubRegionFilterSensitiveLog = SubRegionFilterSensitiveLog;
const AddressFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Label && { Label: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Country && { Country: (0, exports.CountryFilterSensitiveLog)(obj.Country) }),
    ...(obj.Region && { Region: (0, exports.RegionFilterSensitiveLog)(obj.Region) }),
    ...(obj.SubRegion && { SubRegion: (0, exports.SubRegionFilterSensitiveLog)(obj.SubRegion) }),
    ...(obj.Locality && { Locality: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.District && { District: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.SubDistrict && { SubDistrict: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.PostalCode && { PostalCode: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Block && { Block: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.SubBlock && { SubBlock: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Street && { Street: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.StreetComponents && {
        StreetComponents: obj.StreetComponents.map((item) => (0, exports.StreetComponentsFilterSensitiveLog)(item)),
    }),
    ...(obj.AddressNumber && { AddressNumber: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Building && { Building: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.SecondaryAddressComponents && {
        SecondaryAddressComponents: obj.SecondaryAddressComponents.map((item) => (0, exports.SecondaryAddressComponentFilterSensitiveLog)(item)),
    }),
});
exports.AddressFilterSensitiveLog = AddressFilterSensitiveLog;
const PhonemeTranscriptionFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Value && { Value: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Preferred && { Preferred: smithy_client_1.SENSITIVE_STRING }),
});
exports.PhonemeTranscriptionFilterSensitiveLog = PhonemeTranscriptionFilterSensitiveLog;
const AddressComponentPhonemesFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Country && { Country: obj.Country.map((item) => (0, exports.PhonemeTranscriptionFilterSensitiveLog)(item)) }),
    ...(obj.Region && { Region: obj.Region.map((item) => (0, exports.PhonemeTranscriptionFilterSensitiveLog)(item)) }),
    ...(obj.SubRegion && { SubRegion: obj.SubRegion.map((item) => (0, exports.PhonemeTranscriptionFilterSensitiveLog)(item)) }),
    ...(obj.Locality && { Locality: obj.Locality.map((item) => (0, exports.PhonemeTranscriptionFilterSensitiveLog)(item)) }),
    ...(obj.District && { District: obj.District.map((item) => (0, exports.PhonemeTranscriptionFilterSensitiveLog)(item)) }),
    ...(obj.SubDistrict && { SubDistrict: obj.SubDistrict.map((item) => (0, exports.PhonemeTranscriptionFilterSensitiveLog)(item)) }),
    ...(obj.Block && { Block: obj.Block.map((item) => (0, exports.PhonemeTranscriptionFilterSensitiveLog)(item)) }),
    ...(obj.SubBlock && { SubBlock: obj.SubBlock.map((item) => (0, exports.PhonemeTranscriptionFilterSensitiveLog)(item)) }),
    ...(obj.Street && { Street: obj.Street.map((item) => (0, exports.PhonemeTranscriptionFilterSensitiveLog)(item)) }),
});
exports.AddressComponentPhonemesFilterSensitiveLog = AddressComponentPhonemesFilterSensitiveLog;
const FilterCircleFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Center && { Center: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Radius && { Radius: smithy_client_1.SENSITIVE_STRING }),
});
exports.FilterCircleFilterSensitiveLog = FilterCircleFilterSensitiveLog;
const AutocompleteFilterFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.BoundingBox && { BoundingBox: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Circle && { Circle: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.IncludeCountries && { IncludeCountries: smithy_client_1.SENSITIVE_STRING }),
});
exports.AutocompleteFilterFilterSensitiveLog = AutocompleteFilterFilterSensitiveLog;
const AutocompleteRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.QueryText && { QueryText: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.BiasPosition && { BiasPosition: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Filter && { Filter: (0, exports.AutocompleteFilterFilterSensitiveLog)(obj.Filter) }),
    ...(obj.PoliticalView && { PoliticalView: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Key && { Key: smithy_client_1.SENSITIVE_STRING }),
});
exports.AutocompleteRequestFilterSensitiveLog = AutocompleteRequestFilterSensitiveLog;
const HighlightFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Value && { Value: smithy_client_1.SENSITIVE_STRING }),
});
exports.HighlightFilterSensitiveLog = HighlightFilterSensitiveLog;
const CountryHighlightsFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Code && { Code: obj.Code.map((item) => (0, exports.HighlightFilterSensitiveLog)(item)) }),
    ...(obj.Name && { Name: obj.Name.map((item) => (0, exports.HighlightFilterSensitiveLog)(item)) }),
});
exports.CountryHighlightsFilterSensitiveLog = CountryHighlightsFilterSensitiveLog;
const RegionHighlightsFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Code && { Code: obj.Code.map((item) => (0, exports.HighlightFilterSensitiveLog)(item)) }),
    ...(obj.Name && { Name: obj.Name.map((item) => (0, exports.HighlightFilterSensitiveLog)(item)) }),
});
exports.RegionHighlightsFilterSensitiveLog = RegionHighlightsFilterSensitiveLog;
const SubRegionHighlightsFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Code && { Code: obj.Code.map((item) => (0, exports.HighlightFilterSensitiveLog)(item)) }),
    ...(obj.Name && { Name: obj.Name.map((item) => (0, exports.HighlightFilterSensitiveLog)(item)) }),
});
exports.SubRegionHighlightsFilterSensitiveLog = SubRegionHighlightsFilterSensitiveLog;
const AutocompleteAddressHighlightsFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Label && { Label: obj.Label.map((item) => (0, exports.HighlightFilterSensitiveLog)(item)) }),
    ...(obj.Country && { Country: (0, exports.CountryHighlightsFilterSensitiveLog)(obj.Country) }),
    ...(obj.Region && { Region: (0, exports.RegionHighlightsFilterSensitiveLog)(obj.Region) }),
    ...(obj.SubRegion && { SubRegion: (0, exports.SubRegionHighlightsFilterSensitiveLog)(obj.SubRegion) }),
    ...(obj.Locality && { Locality: obj.Locality.map((item) => (0, exports.HighlightFilterSensitiveLog)(item)) }),
    ...(obj.District && { District: obj.District.map((item) => (0, exports.HighlightFilterSensitiveLog)(item)) }),
    ...(obj.SubDistrict && { SubDistrict: obj.SubDistrict.map((item) => (0, exports.HighlightFilterSensitiveLog)(item)) }),
    ...(obj.Street && { Street: obj.Street.map((item) => (0, exports.HighlightFilterSensitiveLog)(item)) }),
    ...(obj.Block && { Block: obj.Block.map((item) => (0, exports.HighlightFilterSensitiveLog)(item)) }),
    ...(obj.SubBlock && { SubBlock: obj.SubBlock.map((item) => (0, exports.HighlightFilterSensitiveLog)(item)) }),
    ...(obj.Intersection && {
        Intersection: obj.Intersection.map((item) => item.map((item) => (0, exports.HighlightFilterSensitiveLog)(item))),
    }),
    ...(obj.PostalCode && { PostalCode: obj.PostalCode.map((item) => (0, exports.HighlightFilterSensitiveLog)(item)) }),
    ...(obj.AddressNumber && { AddressNumber: obj.AddressNumber.map((item) => (0, exports.HighlightFilterSensitiveLog)(item)) }),
    ...(obj.Building && { Building: obj.Building.map((item) => (0, exports.HighlightFilterSensitiveLog)(item)) }),
});
exports.AutocompleteAddressHighlightsFilterSensitiveLog = AutocompleteAddressHighlightsFilterSensitiveLog;
const AutocompleteHighlightsFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Title && { Title: obj.Title.map((item) => (0, exports.HighlightFilterSensitiveLog)(item)) }),
    ...(obj.Address && { Address: (0, exports.AutocompleteAddressHighlightsFilterSensitiveLog)(obj.Address) }),
});
exports.AutocompleteHighlightsFilterSensitiveLog = AutocompleteHighlightsFilterSensitiveLog;
const AutocompleteResultItemFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.PlaceId && { PlaceId: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.PlaceType && { PlaceType: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Title && { Title: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Address && { Address: (0, exports.AddressFilterSensitiveLog)(obj.Address) }),
    ...(obj.Distance && { Distance: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.PoliticalView && { PoliticalView: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Highlights && { Highlights: (0, exports.AutocompleteHighlightsFilterSensitiveLog)(obj.Highlights) }),
});
exports.AutocompleteResultItemFilterSensitiveLog = AutocompleteResultItemFilterSensitiveLog;
const AutocompleteResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.ResultItems && {
        ResultItems: obj.ResultItems.map((item) => (0, exports.AutocompleteResultItemFilterSensitiveLog)(item)),
    }),
});
exports.AutocompleteResponseFilterSensitiveLog = AutocompleteResponseFilterSensitiveLog;
const BusinessChainFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Name && { Name: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Id && { Id: smithy_client_1.SENSITIVE_STRING }),
});
exports.BusinessChainFilterSensitiveLog = BusinessChainFilterSensitiveLog;
const ContactDetailsFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Label && { Label: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Value && { Value: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Categories && { Categories: obj.Categories.map((item) => (0, exports.CategoryFilterSensitiveLog)(item)) }),
});
exports.ContactDetailsFilterSensitiveLog = ContactDetailsFilterSensitiveLog;
const ContactsFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Phones && { Phones: obj.Phones.map((item) => (0, exports.ContactDetailsFilterSensitiveLog)(item)) }),
    ...(obj.Faxes && { Faxes: obj.Faxes.map((item) => (0, exports.ContactDetailsFilterSensitiveLog)(item)) }),
    ...(obj.Websites && { Websites: obj.Websites.map((item) => (0, exports.ContactDetailsFilterSensitiveLog)(item)) }),
    ...(obj.Emails && { Emails: obj.Emails.map((item) => (0, exports.ContactDetailsFilterSensitiveLog)(item)) }),
});
exports.ContactsFilterSensitiveLog = ContactsFilterSensitiveLog;
const FoodTypeFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.LocalizedName && { LocalizedName: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Id && { Id: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Primary && { Primary: smithy_client_1.SENSITIVE_STRING }),
});
exports.FoodTypeFilterSensitiveLog = FoodTypeFilterSensitiveLog;
const GeocodeFilterFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.IncludeCountries && { IncludeCountries: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.IncludePlaceTypes && { IncludePlaceTypes: smithy_client_1.SENSITIVE_STRING }),
});
exports.GeocodeFilterFilterSensitiveLog = GeocodeFilterFilterSensitiveLog;
const GeocodeQueryComponentsFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Country && { Country: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Region && { Region: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.SubRegion && { SubRegion: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Locality && { Locality: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.District && { District: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Street && { Street: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.AddressNumber && { AddressNumber: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.PostalCode && { PostalCode: smithy_client_1.SENSITIVE_STRING }),
});
exports.GeocodeQueryComponentsFilterSensitiveLog = GeocodeQueryComponentsFilterSensitiveLog;
const GeocodeRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.QueryText && { QueryText: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.QueryComponents && { QueryComponents: (0, exports.GeocodeQueryComponentsFilterSensitiveLog)(obj.QueryComponents) }),
    ...(obj.BiasPosition && { BiasPosition: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Filter && { Filter: (0, exports.GeocodeFilterFilterSensitiveLog)(obj.Filter) }),
    ...(obj.PoliticalView && { PoliticalView: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Key && { Key: smithy_client_1.SENSITIVE_STRING }),
});
exports.GeocodeRequestFilterSensitiveLog = GeocodeRequestFilterSensitiveLog;
const IntersectionFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.PlaceId && { PlaceId: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Title && { Title: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Address && { Address: (0, exports.AddressFilterSensitiveLog)(obj.Address) }),
    ...(obj.Position && { Position: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Distance && { Distance: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.RouteDistance && { RouteDistance: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.MapView && { MapView: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.AccessPoints && { AccessPoints: obj.AccessPoints.map((item) => (0, exports.AccessPointFilterSensitiveLog)(item)) }),
});
exports.IntersectionFilterSensitiveLog = IntersectionFilterSensitiveLog;
const RelatedPlaceFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.PlaceId && { PlaceId: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.PlaceType && { PlaceType: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Title && { Title: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Address && { Address: (0, exports.AddressFilterSensitiveLog)(obj.Address) }),
    ...(obj.Position && { Position: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.AccessPoints && { AccessPoints: obj.AccessPoints.map((item) => (0, exports.AccessPointFilterSensitiveLog)(item)) }),
});
exports.RelatedPlaceFilterSensitiveLog = RelatedPlaceFilterSensitiveLog;
const ParsedQueryComponentFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Value && { Value: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.QueryComponent && { QueryComponent: smithy_client_1.SENSITIVE_STRING }),
});
exports.ParsedQueryComponentFilterSensitiveLog = ParsedQueryComponentFilterSensitiveLog;
const ParsedQuerySecondaryAddressComponentFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Value && { Value: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Number && { Number: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Designator && { Designator: smithy_client_1.SENSITIVE_STRING }),
});
exports.ParsedQuerySecondaryAddressComponentFilterSensitiveLog = ParsedQuerySecondaryAddressComponentFilterSensitiveLog;
const GeocodeParsedQueryAddressComponentsFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Country && { Country: obj.Country.map((item) => (0, exports.ParsedQueryComponentFilterSensitiveLog)(item)) }),
    ...(obj.Region && { Region: obj.Region.map((item) => (0, exports.ParsedQueryComponentFilterSensitiveLog)(item)) }),
    ...(obj.SubRegion && { SubRegion: obj.SubRegion.map((item) => (0, exports.ParsedQueryComponentFilterSensitiveLog)(item)) }),
    ...(obj.Locality && { Locality: obj.Locality.map((item) => (0, exports.ParsedQueryComponentFilterSensitiveLog)(item)) }),
    ...(obj.District && { District: obj.District.map((item) => (0, exports.ParsedQueryComponentFilterSensitiveLog)(item)) }),
    ...(obj.SubDistrict && { SubDistrict: obj.SubDistrict.map((item) => (0, exports.ParsedQueryComponentFilterSensitiveLog)(item)) }),
    ...(obj.PostalCode && { PostalCode: obj.PostalCode.map((item) => (0, exports.ParsedQueryComponentFilterSensitiveLog)(item)) }),
    ...(obj.Block && { Block: obj.Block.map((item) => (0, exports.ParsedQueryComponentFilterSensitiveLog)(item)) }),
    ...(obj.SubBlock && { SubBlock: obj.SubBlock.map((item) => (0, exports.ParsedQueryComponentFilterSensitiveLog)(item)) }),
    ...(obj.Street && { Street: obj.Street.map((item) => (0, exports.ParsedQueryComponentFilterSensitiveLog)(item)) }),
    ...(obj.AddressNumber && {
        AddressNumber: obj.AddressNumber.map((item) => (0, exports.ParsedQueryComponentFilterSensitiveLog)(item)),
    }),
    ...(obj.Building && { Building: obj.Building.map((item) => (0, exports.ParsedQueryComponentFilterSensitiveLog)(item)) }),
    ...(obj.SecondaryAddressComponents && {
        SecondaryAddressComponents: obj.SecondaryAddressComponents.map((item) => (0, exports.ParsedQuerySecondaryAddressComponentFilterSensitiveLog)(item)),
    }),
});
exports.GeocodeParsedQueryAddressComponentsFilterSensitiveLog = GeocodeParsedQueryAddressComponentsFilterSensitiveLog;
const GeocodeParsedQueryFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Title && { Title: obj.Title.map((item) => (0, exports.ParsedQueryComponentFilterSensitiveLog)(item)) }),
    ...(obj.Address && { Address: (0, exports.GeocodeParsedQueryAddressComponentsFilterSensitiveLog)(obj.Address) }),
});
exports.GeocodeParsedQueryFilterSensitiveLog = GeocodeParsedQueryFilterSensitiveLog;
const UspsZipFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.ZipClassificationCode && { ZipClassificationCode: smithy_client_1.SENSITIVE_STRING }),
});
exports.UspsZipFilterSensitiveLog = UspsZipFilterSensitiveLog;
const UspsZipPlus4FilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.RecordTypeCode && { RecordTypeCode: smithy_client_1.SENSITIVE_STRING }),
});
exports.UspsZipPlus4FilterSensitiveLog = UspsZipPlus4FilterSensitiveLog;
const PostalCodeDetailsFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.PostalCode && { PostalCode: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.PostalAuthority && { PostalAuthority: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.PostalCodeType && { PostalCodeType: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.UspsZip && { UspsZip: (0, exports.UspsZipFilterSensitiveLog)(obj.UspsZip) }),
    ...(obj.UspsZipPlus4 && { UspsZipPlus4: (0, exports.UspsZipPlus4FilterSensitiveLog)(obj.UspsZipPlus4) }),
});
exports.PostalCodeDetailsFilterSensitiveLog = PostalCodeDetailsFilterSensitiveLog;
const TimeZoneFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Name && { Name: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Offset && { Offset: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.OffsetSeconds && { OffsetSeconds: smithy_client_1.SENSITIVE_STRING }),
});
exports.TimeZoneFilterSensitiveLog = TimeZoneFilterSensitiveLog;
const GeocodeResultItemFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.PlaceId && { PlaceId: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.PlaceType && { PlaceType: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Title && { Title: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Address && { Address: (0, exports.AddressFilterSensitiveLog)(obj.Address) }),
    ...(obj.AddressNumberCorrected && { AddressNumberCorrected: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.PostalCodeDetails && {
        PostalCodeDetails: obj.PostalCodeDetails.map((item) => (0, exports.PostalCodeDetailsFilterSensitiveLog)(item)),
    }),
    ...(obj.Position && { Position: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Distance && { Distance: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.MapView && { MapView: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Categories && { Categories: obj.Categories.map((item) => (0, exports.CategoryFilterSensitiveLog)(item)) }),
    ...(obj.FoodTypes && { FoodTypes: obj.FoodTypes.map((item) => (0, exports.FoodTypeFilterSensitiveLog)(item)) }),
    ...(obj.AccessPoints && { AccessPoints: obj.AccessPoints.map((item) => (0, exports.AccessPointFilterSensitiveLog)(item)) }),
    ...(obj.TimeZone && { TimeZone: (0, exports.TimeZoneFilterSensitiveLog)(obj.TimeZone) }),
    ...(obj.PoliticalView && { PoliticalView: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.ParsedQuery && { ParsedQuery: (0, exports.GeocodeParsedQueryFilterSensitiveLog)(obj.ParsedQuery) }),
    ...(obj.Intersections && { Intersections: obj.Intersections.map((item) => (0, exports.IntersectionFilterSensitiveLog)(item)) }),
    ...(obj.MainAddress && { MainAddress: (0, exports.RelatedPlaceFilterSensitiveLog)(obj.MainAddress) }),
    ...(obj.SecondaryAddresses && {
        SecondaryAddresses: obj.SecondaryAddresses.map((item) => (0, exports.RelatedPlaceFilterSensitiveLog)(item)),
    }),
});
exports.GeocodeResultItemFilterSensitiveLog = GeocodeResultItemFilterSensitiveLog;
const GeocodeResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.ResultItems && { ResultItems: obj.ResultItems.map((item) => (0, exports.GeocodeResultItemFilterSensitiveLog)(item)) }),
});
exports.GeocodeResponseFilterSensitiveLog = GeocodeResponseFilterSensitiveLog;
const GetPlaceRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.PlaceId && { PlaceId: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.PoliticalView && { PoliticalView: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Key && { Key: smithy_client_1.SENSITIVE_STRING }),
});
exports.GetPlaceRequestFilterSensitiveLog = GetPlaceRequestFilterSensitiveLog;
const OpeningHoursComponentsFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.OpenTime && { OpenTime: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.OpenDuration && { OpenDuration: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Recurrence && { Recurrence: smithy_client_1.SENSITIVE_STRING }),
});
exports.OpeningHoursComponentsFilterSensitiveLog = OpeningHoursComponentsFilterSensitiveLog;
const OpeningHoursFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Display && { Display: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.OpenNow && { OpenNow: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Components && { Components: obj.Components.map((item) => (0, exports.OpeningHoursComponentsFilterSensitiveLog)(item)) }),
    ...(obj.Categories && { Categories: obj.Categories.map((item) => (0, exports.CategoryFilterSensitiveLog)(item)) }),
});
exports.OpeningHoursFilterSensitiveLog = OpeningHoursFilterSensitiveLog;
const PhonemeDetailsFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Title && { Title: obj.Title.map((item) => (0, exports.PhonemeTranscriptionFilterSensitiveLog)(item)) }),
    ...(obj.Address && { Address: (0, exports.AddressComponentPhonemesFilterSensitiveLog)(obj.Address) }),
});
exports.PhonemeDetailsFilterSensitiveLog = PhonemeDetailsFilterSensitiveLog;
const GetPlaceResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.PlaceId && { PlaceId: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.PlaceType && { PlaceType: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Title && { Title: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Address && { Address: (0, exports.AddressFilterSensitiveLog)(obj.Address) }),
    ...(obj.AddressNumberCorrected && { AddressNumberCorrected: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.PostalCodeDetails && {
        PostalCodeDetails: obj.PostalCodeDetails.map((item) => (0, exports.PostalCodeDetailsFilterSensitiveLog)(item)),
    }),
    ...(obj.Position && { Position: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.MapView && { MapView: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Categories && { Categories: obj.Categories.map((item) => (0, exports.CategoryFilterSensitiveLog)(item)) }),
    ...(obj.FoodTypes && { FoodTypes: obj.FoodTypes.map((item) => (0, exports.FoodTypeFilterSensitiveLog)(item)) }),
    ...(obj.BusinessChains && {
        BusinessChains: obj.BusinessChains.map((item) => (0, exports.BusinessChainFilterSensitiveLog)(item)),
    }),
    ...(obj.Contacts && { Contacts: (0, exports.ContactsFilterSensitiveLog)(obj.Contacts) }),
    ...(obj.OpeningHours && { OpeningHours: obj.OpeningHours.map((item) => (0, exports.OpeningHoursFilterSensitiveLog)(item)) }),
    ...(obj.AccessPoints && { AccessPoints: obj.AccessPoints.map((item) => (0, exports.AccessPointFilterSensitiveLog)(item)) }),
    ...(obj.AccessRestrictions && {
        AccessRestrictions: obj.AccessRestrictions.map((item) => (0, exports.AccessRestrictionFilterSensitiveLog)(item)),
    }),
    ...(obj.TimeZone && { TimeZone: (0, exports.TimeZoneFilterSensitiveLog)(obj.TimeZone) }),
    ...(obj.PoliticalView && { PoliticalView: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Phonemes && { Phonemes: (0, exports.PhonemeDetailsFilterSensitiveLog)(obj.Phonemes) }),
    ...(obj.MainAddress && { MainAddress: (0, exports.RelatedPlaceFilterSensitiveLog)(obj.MainAddress) }),
    ...(obj.SecondaryAddresses && {
        SecondaryAddresses: obj.SecondaryAddresses.map((item) => (0, exports.RelatedPlaceFilterSensitiveLog)(item)),
    }),
});
exports.GetPlaceResponseFilterSensitiveLog = GetPlaceResponseFilterSensitiveLog;
const ReverseGeocodeRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.QueryPosition && { QueryPosition: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.QueryRadius && { QueryRadius: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.PoliticalView && { PoliticalView: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Key && { Key: smithy_client_1.SENSITIVE_STRING }),
});
exports.ReverseGeocodeRequestFilterSensitiveLog = ReverseGeocodeRequestFilterSensitiveLog;
const ReverseGeocodeResultItemFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.PlaceId && { PlaceId: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.PlaceType && { PlaceType: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Title && { Title: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Address && { Address: (0, exports.AddressFilterSensitiveLog)(obj.Address) }),
    ...(obj.AddressNumberCorrected && { AddressNumberCorrected: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.PostalCodeDetails && {
        PostalCodeDetails: obj.PostalCodeDetails.map((item) => (0, exports.PostalCodeDetailsFilterSensitiveLog)(item)),
    }),
    ...(obj.Position && { Position: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Distance && { Distance: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.MapView && { MapView: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Categories && { Categories: obj.Categories.map((item) => (0, exports.CategoryFilterSensitiveLog)(item)) }),
    ...(obj.FoodTypes && { FoodTypes: obj.FoodTypes.map((item) => (0, exports.FoodTypeFilterSensitiveLog)(item)) }),
    ...(obj.AccessPoints && { AccessPoints: obj.AccessPoints.map((item) => (0, exports.AccessPointFilterSensitiveLog)(item)) }),
    ...(obj.TimeZone && { TimeZone: (0, exports.TimeZoneFilterSensitiveLog)(obj.TimeZone) }),
    ...(obj.PoliticalView && { PoliticalView: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Intersections && { Intersections: obj.Intersections.map((item) => (0, exports.IntersectionFilterSensitiveLog)(item)) }),
});
exports.ReverseGeocodeResultItemFilterSensitiveLog = ReverseGeocodeResultItemFilterSensitiveLog;
const ReverseGeocodeResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.ResultItems && {
        ResultItems: obj.ResultItems.map((item) => (0, exports.ReverseGeocodeResultItemFilterSensitiveLog)(item)),
    }),
});
exports.ReverseGeocodeResponseFilterSensitiveLog = ReverseGeocodeResponseFilterSensitiveLog;
const SearchNearbyFilterFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.BoundingBox && { BoundingBox: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.IncludeCountries && { IncludeCountries: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.IncludeCategories && { IncludeCategories: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.ExcludeCategories && { ExcludeCategories: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.IncludeBusinessChains && { IncludeBusinessChains: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.ExcludeBusinessChains && { ExcludeBusinessChains: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.IncludeFoodTypes && { IncludeFoodTypes: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.ExcludeFoodTypes && { ExcludeFoodTypes: smithy_client_1.SENSITIVE_STRING }),
});
exports.SearchNearbyFilterFilterSensitiveLog = SearchNearbyFilterFilterSensitiveLog;
const SearchNearbyRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.QueryPosition && { QueryPosition: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.QueryRadius && { QueryRadius: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Filter && { Filter: (0, exports.SearchNearbyFilterFilterSensitiveLog)(obj.Filter) }),
    ...(obj.PoliticalView && { PoliticalView: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Key && { Key: smithy_client_1.SENSITIVE_STRING }),
});
exports.SearchNearbyRequestFilterSensitiveLog = SearchNearbyRequestFilterSensitiveLog;
const SearchNearbyResultItemFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.PlaceId && { PlaceId: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.PlaceType && { PlaceType: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Title && { Title: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Address && { Address: (0, exports.AddressFilterSensitiveLog)(obj.Address) }),
    ...(obj.AddressNumberCorrected && { AddressNumberCorrected: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Position && { Position: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Distance && { Distance: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.MapView && { MapView: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Categories && { Categories: obj.Categories.map((item) => (0, exports.CategoryFilterSensitiveLog)(item)) }),
    ...(obj.FoodTypes && { FoodTypes: obj.FoodTypes.map((item) => (0, exports.FoodTypeFilterSensitiveLog)(item)) }),
    ...(obj.BusinessChains && {
        BusinessChains: obj.BusinessChains.map((item) => (0, exports.BusinessChainFilterSensitiveLog)(item)),
    }),
    ...(obj.Contacts && { Contacts: (0, exports.ContactsFilterSensitiveLog)(obj.Contacts) }),
    ...(obj.OpeningHours && { OpeningHours: obj.OpeningHours.map((item) => (0, exports.OpeningHoursFilterSensitiveLog)(item)) }),
    ...(obj.AccessPoints && { AccessPoints: obj.AccessPoints.map((item) => (0, exports.AccessPointFilterSensitiveLog)(item)) }),
    ...(obj.AccessRestrictions && {
        AccessRestrictions: obj.AccessRestrictions.map((item) => (0, exports.AccessRestrictionFilterSensitiveLog)(item)),
    }),
    ...(obj.TimeZone && { TimeZone: (0, exports.TimeZoneFilterSensitiveLog)(obj.TimeZone) }),
    ...(obj.PoliticalView && { PoliticalView: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Phonemes && { Phonemes: (0, exports.PhonemeDetailsFilterSensitiveLog)(obj.Phonemes) }),
});
exports.SearchNearbyResultItemFilterSensitiveLog = SearchNearbyResultItemFilterSensitiveLog;
const SearchNearbyResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.ResultItems && {
        ResultItems: obj.ResultItems.map((item) => (0, exports.SearchNearbyResultItemFilterSensitiveLog)(item)),
    }),
});
exports.SearchNearbyResponseFilterSensitiveLog = SearchNearbyResponseFilterSensitiveLog;
const SearchTextFilterFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.BoundingBox && { BoundingBox: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Circle && { Circle: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.IncludeCountries && { IncludeCountries: smithy_client_1.SENSITIVE_STRING }),
});
exports.SearchTextFilterFilterSensitiveLog = SearchTextFilterFilterSensitiveLog;
const SearchTextRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.QueryText && { QueryText: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.QueryId && { QueryId: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.BiasPosition && { BiasPosition: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Filter && { Filter: (0, exports.SearchTextFilterFilterSensitiveLog)(obj.Filter) }),
    ...(obj.PoliticalView && { PoliticalView: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Key && { Key: smithy_client_1.SENSITIVE_STRING }),
});
exports.SearchTextRequestFilterSensitiveLog = SearchTextRequestFilterSensitiveLog;
const SearchTextResultItemFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.PlaceId && { PlaceId: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.PlaceType && { PlaceType: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Title && { Title: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Address && { Address: (0, exports.AddressFilterSensitiveLog)(obj.Address) }),
    ...(obj.AddressNumberCorrected && { AddressNumberCorrected: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Position && { Position: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Distance && { Distance: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.MapView && { MapView: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Categories && { Categories: obj.Categories.map((item) => (0, exports.CategoryFilterSensitiveLog)(item)) }),
    ...(obj.FoodTypes && { FoodTypes: obj.FoodTypes.map((item) => (0, exports.FoodTypeFilterSensitiveLog)(item)) }),
    ...(obj.BusinessChains && {
        BusinessChains: obj.BusinessChains.map((item) => (0, exports.BusinessChainFilterSensitiveLog)(item)),
    }),
    ...(obj.Contacts && { Contacts: (0, exports.ContactsFilterSensitiveLog)(obj.Contacts) }),
    ...(obj.OpeningHours && { OpeningHours: obj.OpeningHours.map((item) => (0, exports.OpeningHoursFilterSensitiveLog)(item)) }),
    ...(obj.AccessPoints && { AccessPoints: obj.AccessPoints.map((item) => (0, exports.AccessPointFilterSensitiveLog)(item)) }),
    ...(obj.AccessRestrictions && {
        AccessRestrictions: obj.AccessRestrictions.map((item) => (0, exports.AccessRestrictionFilterSensitiveLog)(item)),
    }),
    ...(obj.TimeZone && { TimeZone: (0, exports.TimeZoneFilterSensitiveLog)(obj.TimeZone) }),
    ...(obj.PoliticalView && { PoliticalView: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Phonemes && { Phonemes: (0, exports.PhonemeDetailsFilterSensitiveLog)(obj.Phonemes) }),
});
exports.SearchTextResultItemFilterSensitiveLog = SearchTextResultItemFilterSensitiveLog;
const SearchTextResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.ResultItems && { ResultItems: obj.ResultItems.map((item) => (0, exports.SearchTextResultItemFilterSensitiveLog)(item)) }),
});
exports.SearchTextResponseFilterSensitiveLog = SearchTextResponseFilterSensitiveLog;
const SuggestFilterFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.BoundingBox && { BoundingBox: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Circle && { Circle: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.IncludeCountries && { IncludeCountries: smithy_client_1.SENSITIVE_STRING }),
});
exports.SuggestFilterFilterSensitiveLog = SuggestFilterFilterSensitiveLog;
const SuggestRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.QueryText && { QueryText: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.BiasPosition && { BiasPosition: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Filter && { Filter: (0, exports.SuggestFilterFilterSensitiveLog)(obj.Filter) }),
    ...(obj.PoliticalView && { PoliticalView: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Key && { Key: smithy_client_1.SENSITIVE_STRING }),
});
exports.SuggestRequestFilterSensitiveLog = SuggestRequestFilterSensitiveLog;
const QueryRefinementFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.RefinedTerm && { RefinedTerm: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.OriginalTerm && { OriginalTerm: smithy_client_1.SENSITIVE_STRING }),
});
exports.QueryRefinementFilterSensitiveLog = QueryRefinementFilterSensitiveLog;
const SuggestAddressHighlightsFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Label && { Label: obj.Label.map((item) => (0, exports.HighlightFilterSensitiveLog)(item)) }),
});
exports.SuggestAddressHighlightsFilterSensitiveLog = SuggestAddressHighlightsFilterSensitiveLog;
const SuggestHighlightsFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Title && { Title: obj.Title.map((item) => (0, exports.HighlightFilterSensitiveLog)(item)) }),
    ...(obj.Address && { Address: (0, exports.SuggestAddressHighlightsFilterSensitiveLog)(obj.Address) }),
});
exports.SuggestHighlightsFilterSensitiveLog = SuggestHighlightsFilterSensitiveLog;
const SuggestPlaceResultFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.PlaceId && { PlaceId: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.PlaceType && { PlaceType: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Address && { Address: (0, exports.AddressFilterSensitiveLog)(obj.Address) }),
    ...(obj.Position && { Position: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Distance && { Distance: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.MapView && { MapView: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Categories && { Categories: obj.Categories.map((item) => (0, exports.CategoryFilterSensitiveLog)(item)) }),
    ...(obj.FoodTypes && { FoodTypes: obj.FoodTypes.map((item) => (0, exports.FoodTypeFilterSensitiveLog)(item)) }),
    ...(obj.BusinessChains && {
        BusinessChains: obj.BusinessChains.map((item) => (0, exports.BusinessChainFilterSensitiveLog)(item)),
    }),
    ...(obj.AccessPoints && { AccessPoints: obj.AccessPoints.map((item) => (0, exports.AccessPointFilterSensitiveLog)(item)) }),
    ...(obj.AccessRestrictions && {
        AccessRestrictions: obj.AccessRestrictions.map((item) => (0, exports.AccessRestrictionFilterSensitiveLog)(item)),
    }),
    ...(obj.TimeZone && { TimeZone: (0, exports.TimeZoneFilterSensitiveLog)(obj.TimeZone) }),
    ...(obj.PoliticalView && { PoliticalView: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Phonemes && { Phonemes: (0, exports.PhonemeDetailsFilterSensitiveLog)(obj.Phonemes) }),
});
exports.SuggestPlaceResultFilterSensitiveLog = SuggestPlaceResultFilterSensitiveLog;
const SuggestQueryResultFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.QueryId && { QueryId: smithy_client_1.SENSITIVE_STRING }),
});
exports.SuggestQueryResultFilterSensitiveLog = SuggestQueryResultFilterSensitiveLog;
const SuggestResultItemFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Title && { Title: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.Place && { Place: (0, exports.SuggestPlaceResultFilterSensitiveLog)(obj.Place) }),
    ...(obj.Query && { Query: (0, exports.SuggestQueryResultFilterSensitiveLog)(obj.Query) }),
    ...(obj.Highlights && { Highlights: (0, exports.SuggestHighlightsFilterSensitiveLog)(obj.Highlights) }),
});
exports.SuggestResultItemFilterSensitiveLog = SuggestResultItemFilterSensitiveLog;
const SuggestResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.ResultItems && { ResultItems: obj.ResultItems.map((item) => (0, exports.SuggestResultItemFilterSensitiveLog)(item)) }),
    ...(obj.QueryRefinements && {
        QueryRefinements: obj.QueryRefinements.map((item) => (0, exports.QueryRefinementFilterSensitiveLog)(item)),
    }),
});
exports.SuggestResponseFilterSensitiveLog = SuggestResponseFilterSensitiveLog;
