// @aws-sdk/client-geo-maps/dist-cjs/GeoMaps.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GeoMaps = void 0;
const smithy_client_1 = require("@smithy/smithy-client");
const GetGlyphsCommand_1 = require("./commands/GetGlyphsCommand");
const GetSpritesCommand_1 = require("./commands/GetSpritesCommand");
const GetStaticMapCommand_1 = require("./commands/GetStaticMapCommand");
const GetStyleDescriptorCommand_1 = require("./commands/GetStyleDescriptorCommand");
const GetTileCommand_1 = require("./commands/GetTileCommand");
const GeoMapsClient_1 = require("./GeoMapsClient");
const commands = {
    GetGlyphsCommand: GetGlyphsCommand_1.GetGlyphsCommand,
    GetSpritesCommand: GetSpritesCommand_1.GetSpritesCommand,
    GetStaticMapCommand: GetStaticMapCommand_1.GetStaticMapCommand,
    GetStyleDescriptorCommand: GetStyleDescriptorCommand_1.GetStyleDescriptorCommand,
    GetTileCommand: GetTileCommand_1.GetTileCommand,
};
class GeoMaps extends GeoMapsClient_1.GeoMapsClient {
}
exports.GeoMaps = GeoMaps;
(0, smithy_client_1.createAggregatedClient)(commands, GeoMaps);
